frontend/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -3,26 +3,33 @@ import {decodeToken, scheduleTokenRefresh} from '../../services/authService';
 import {useNavigate} from 'react-router-dom';
 import {message} from 'antd';
 import * as Constants from "../../consts/consts.ts";
+import { getAuthStatus } from "../../api/auth/authStatus.ts";
+import { UserStatus } from "../../types/authTypes.ts";
 
 type AuthContextType = {
     isLoggedIn: boolean;
     user: { [key: string]: any } | null;
+    userStatus: UserStatus | null;
```

Comment:
I’ll leave the decision up to you, but I think it would be better to ignore the user part in user_status and just save the permissions. It would make everything clearer.

Reply:
yes